backend/app/api/events.py

```diff
@@ -54,10 +56,78 @@ def query_eventbrite(city: str, date: str, keywords: List[str]) -> List[Event]:
+            start = current_date.replace(hour=0, minute=0, second=0, microsecond=0)+ timedelta(days=1)
+            end = current_date.replace(hour=23, minute=59, second=59, microsecond=0)+ timedelta(days=1)
+            return start.isoformat(), end.isoformat()
+        else:
+            # Handle YYYY-MM-DD format
+            try:
+                parsed_date = datetime.strptime(date_entity, "%Y-%m-%d")
+                start_date = parsed_date.replace(hour=0, minute=0, second=0, microsecond=0)
+                end_date = parsed_date.replace(hour=23, minute=59, second=59, microsecond=0)
+                return start_date.isoformat(), end_date.isoformat()
+            except ValueError:
+                # Handle specific formats like "23-11" (dd-mm)
+                try:
+                    parsed_date = datetime.strptime(date_entity, "%d-%m").replace(year=current_date.year)
+                    start_date = parsed_date.replace(hour=0, minute=0, second=0, microsecond=0)
+                    end_date = parsed_date.replace(hour=23, minute=59, second=59, microsecond=0)
+                    return start_date.isoformat(), end_date.isoformat()
+                except ValueError:
+                    return None, None
+                
+def query_ticketmaster(city: str, date: str, keywords: List[str]) -> List[Event]:
+    """Fetch events from Ticketmaster API."""
+    # Remove the date and city from keywords to avoid confusion using API
+    filtered_keywords = [kw for kw in keywords if kw.lower() != city.lower() and kw.lower() != date.lower()]
+    # Set the day start and end, handle cases like "tomorrow", "today"
+    date, end_date = get_formatted_date(date)
+    print(f"Querying Ticketmaster: city={city}, date={date}, endDate={end_date}, keywords={filtered_keywords}")
+
+    params = {
+        "keyword": " ".join(filtered_keywords),
+        "city": f"{city}",
+        "sort": "date,asc",
+        "startDateTime": date + "+00:00",
+        "endDateTime": end_date + "+00:00",
+        "apikey": TICKETMASTER_API_KEY,
+        "size": 5 #page size of the response
+        }
+    response = requests.get(TICKETMASTER_API_URL, params=params)
+    if response.status_code != 200:
+        print(f"TicketMaster API Error: {response.status_code} {response.text}")
+        return []
+    
+    # Parse event data
+    events_data = response.json().get("_embedded", {}).get("events", [])
+    events = []
+    
+    for event_data in events_data:
+        event = Event(
+            id=event_data.get("id"),
+            name=event_data.get("name"),
+            location=city,  # Event location set to the input city
+            url=event_data.get("url"),
+            date=event_data["dates"]["start"]["localDate"],
+            category = event_data.get("classifications", [])[0].get("segment", {}).get("name", "Unknown Category")
+        )
+        events.append(event)
+    if events == []:
+        return "No events found."
```

Comment:
Everywhere you return a list, yet here you return `string`? Smells like a bug

Reply:
fixed